share/sysbuild/CMakeLists.txt

```diff
@@ -35,7 +35,9 @@ get_property(IMAGES GLOBAL PROPERTY sysbuild_images)
 sysbuild_module_call(PRE_CMAKE MODULES ${SYSBUILD_MODULE_NAMES} IMAGES ${IMAGES})
 sysbuild_images_order(IMAGES_CONFIGURATION_ORDER CONFIGURE IMAGES ${IMAGES})
 foreach(image ${IMAGES_CONFIGURATION_ORDER})
+  sysbuild_module_call(PRE_IMAGE_CMAKE MODULES ${SYSBUILD_MODULE_NAMES} IMAGES ${IMAGES} IMAGE ${image})
```

Comment:
If we go for these hooks, then I believe they should be defined per image, not per module. In other words, instead of a `${SYSBUILD_CURRENT_MODULE_NAME}_pre_image_cmake` function, support having a `${image}_pre_image_cmake`.

Reply:
I don't agree with that, whilst in this instance it is mcuboot which has the hook, the hook could very well reside in a module, it could be needed for multiple applications not just one, therefore making an image name specific hook is much more cumbersome

Reply:
I think it would be more convenient to add a separate hook function for each new image, instead of hunting down and patching an existing, module-specific hook. If multiple images need to share the same hook, then the common code could be moved into some macro, called from multiple functions.

Decoupling this from modules would also open up the possibility of adding hooks for images within Zephyr.

Reply:
> I think it would be more convenient to add a separate hook function for each new image, instead of hunting down and patching an existing, module-specific hook. If multiple images need to share the same hook, then the common code could be moved into some macro, called from multiple functions.

But you can't then do that, how do you have a single function that runs for **all** images after each one has been configured when you don't know the possible names of the applications or want to accept any application name?

Reply:
I see. I can't think of a use case for **all** images having a shared `POST_IMAGE_CMAKE` hook, but if that's important, then using modules is fine.

Reply:
while `pre` and `post` hooks may be convenient and more visible to users, then I just want to point out the fact that because those hooks surrounds a single function, that is: `ExternalZephyrProject_Cmake(APPLICATION ${image})`
then CMake already provides a mechanism for this.

Any module can redeclare the `ExternalZephyrProject_Cmake(APPLICATION ${image})` function like this, and add their hooks.

For example in Zephyr module `foo` then do this:
```
# foo/sysbuild/CMakeLists.txt

function(ExternalZephyrProject_Cmake)
    # Call any prehook functions.
    prehook_functions()
  
  # Call original implementation.
   _ExternalZephyrProject_Cmake(${ARGN})

  # Call any posthook functions.
  posthook_functions()
endfunction()
```
but as said, I'm fine with an official and documented sysbuild feature for this.

> I think it would be more convenient to add a separate hook function for each new image

but that we have already for the `prehook` part through the IMAGE_CONF_SCRIPT property.
So anywhere before CMake is invoked for the images then  you can do the following:
```
set_property(TARGET <image> APPEND PROPERTY IMAGE_CONF_SCRIPT <myscript.cmake>)
```
Result, before CMake is invoked on `<image>`, then `<myscript.cmake>` is included.

Reply:
> but that we have already for the prehook part through the IMAGE_CONF_SCRIPT property.

This has a default set for images already which would then prevent the default from running, so it's not really usable like that